### PR TITLE
Fixes to be able to run upon windows XP.

### PR DESCRIPTION
**Pull request by goodok**
It is related with the lines like 'cmd ("cd %s; git clone ...")' that is when a few commands run as one line.

In windows XP the separator must be strictly "&&" (or "&"), but not ";". 
Aaron remarks: && is better and for Unix too, as it will prevent execution of the second command if the first fails.

The second commit is related with output encodings. Author's name are sometimes unicode string, so in the  windows terminal (when it is not utf8) they must be encoded properly.

### CODE REVIEW

**Comment by goodok** [comment]
@smichr Please check it upon your Windows machine.

**Comment by asmeurer** [comment]
I'm using this to run tests on sympy/sympy#933.  If it works (i.e., uploads fine), then I am +1.

**Comment by smichr** [comment]
can you push this to a sympy branch instead of a sympy-bot branch? I'm not sure of the address to get this branch, but I do have your other branches. (Sorry if the question is confusing; look at the URL https://github.com/sympy/sympy-bot/pull/86 -- it's at sympy-bot not plain sympy.)

**Comment by smichr** [comment]
OK, I see. This is not the py3k related work, this has to do with the sympy-bot...which I have never used. If you can point me to what to try, I'll try it.

**Comment by goodok** [comment]
No, I mean that you can check the running of 'sympy-bot' upon windows (If you use it earlier, of course), not sympy.
Sympy-bot have not worked upon windows, so I addressed to you to check it.
But if you used sympy-bot earler

**Comment by goodok** [comment]
You can try to clone sympy-bot and run this commands in master branch and in this branch.

```
> python sympy-bot list
> python sympy-bot review -n 938     # (or another number)
```

It willl be good if for this branch there are no errors. (like they are ecpected in master)

**Comment by smichr** [comment]
On Sat, Jan 7, 2012 at 9:55 PM, Alexey U. Gudchenko <
reply@reply.github.com

> wrote:
> 
> No, I mean that you can check the running of 'sympy-bot' upon windows (If
> you use it earlier, of course), not sympy.
> Sympy-bot have not worked upon windows, so I addressed to you to check it.
> But if you used sympy-bot earler
> 
> Nope, never have. Maybe someone else can give it a try?

**Comment by smichr** [comment]
OK, here's what I did and what I got:

$ git clone git@github.com:sympy/sympy-bot.git
Cloning into sympy-bot...
remote: Counting objects: 696, done.
remote: Compressing objects: 100% (301/301), done.
remote: Total 696 (delta 418), reused 649 (delta 378)
Receiving objects: 100% (696/696), 167.39 KiB | 56 KiB/s, done.
Resolving deltas: 100% (418/418), done.

chriss@CHRIS-LT ~/sympy (ga)
$ cd sympy-bot

chriss@CHRIS-LT ~/sympy/sympy-bot (master)
$ c:/python27/python.exe sympy-bot list
Patches that cannot be merged without conflicts:
#233: https://github.com/goodok/sympy 101_wikitest
      Author   : "Alexey U. Gudchenko" <>
      Date     : Sun Apr 17 22:04:49 2011
#239: https://github.com/sherjilozair/sympy eigenvalsort
      Author   : "Sherjil Ozair" sherjilozair@gmail.com
      Date     : Wed Apr 20 08:01:06 2011
#268: https://github.com/certik/sympy tableform_array
Traceback (most recent call last):
  File "sympy-bot", line 316, in <module>
    main()
  File "sympy-bot", line 90, in main
    list_pull_requests(options.repository)
  File "c:\Users\cps\sympy\sympy-bot\utils.py", line 212, in
list_pull_reques
ts
    print "      Author   : %s" % author
  File "c:\python27\lib\encodings\cp437.py", line 12, in encode
    return codecs.charmap_encode(input,errors,encoding_map)
UnicodeEncodeError: 'charmap' codec can't encode character u'\u0159' in
position
 21: character maps to <undefined>

chriss@CHRIS-LT ~/sympy/sympy-bot (master)
$ c:/python27/python.exe sympy-bot review -n 938

> Working directory: c:\users\cps\appdata\local\temp\sympy-bot-tmpb9tffz
> Reviewing pull request #938
> Pull request info:
>     Author: "unknown" <>
>     Repository: https://github.com/raoulb/sympy
>     Branch: fresnel
> Cloning sympy/sympy master
> The filename, directory name, or volume label syntax is incorrect.
> Traceback (most recent call last):
>   File "sympy-bot", line 316, in <module>
>     main()
>   File "sympy-bot", line 126, in main
>     review(n, options, username, password)
>   File "sympy-bot", line 261, in review
>     config.repository) % tmpdir)
>   File "c:\Users\cps\sympy\sympy-bot\utils.py", line 49, in cmd
>     raise CmdException("Command '%s' failed with err=%d." % (s, r))
> utils.CmdException: Command 'cd
> c:\users\cps\appdata\local\temp\sympy-bot-tmp
> b9tffz; git clone git://github.com/sympy/sympy.git' failed with err=1.

/c

**Comment by smichr** [comment]
I copied what you have here from the raw file display of github and here are the results of the first command:

$ c:/python27/python.exe sympy-bot list
Patches that cannot be merged without conflicts:
#233: https://github.com/goodok/sympy 101_wikitest
      Author   : "Alexey U. Gudchenko" <>
      Date     : Sun Apr 17 22:04:49 2011
#239: https://github.com/sherjilozair/sympy eigenvalsort
      Author   : "Sherjil Ozair" sherjilozair@gmail.com
      Date     : Wed Apr 20 08:01:06 2011
#268: https://github.com/certik/sympy tableform_array
      Author   : "Ond┼Öej ─îert├¡k" ondrej@certik.cz
      Date     : Thu Apr 28 05:42:05 2011
#343: https://github.com/rlamy/sympy ask
      Author   : "Ronan Lamy" <>
      Date     : Sat May 21 04:29:47 2011
#359: https://github.com/rlamy/sympy subclass
      Author   : "Ronan Lamy" <>
      Date     : Wed May 25 05:27:31 2011
#389: https://github.com/rlamy/sympy print-Lambda
      Author   : "Ronan Lamy" <>
      Date     : Mon Jun 06 18:40:00 2011
#444: https://github.com/sherjilozair/sympy type
      Author   : "Sherjil Ozair" sherjilozair@gmail.com
      Date     : Tue Jun 21 13:23:00 2011
#447: https://github.com/saptman/sympy partitions
      Author   : "Saptarshi Mandal" sapta.iitkgp@gmail.com
      Date     : Wed Jun 22 04:12:04 2011
#456: https://github.com/sherjilozair/sympy smallchanges
      Author   : "Sherjil Ozair" sherjilozair@gmail.com
      Date     : Fri Jun 24 13:13:36 2011
#543: https://github.com/ness01/sympy gsoc-3
      Author   : "Tom Bachmann" <>
      Date     : Sun Aug 14 15:40:43 2011
#563: https://github.com/Jerryy/sympy ratsimp4
      Author   : "unknown" <>
      Date     : Thu Aug 18 08:34:08 2011
#573: https://github.com/lazovich/sympy represent2
      Author   : "unknown" <>
      Date     : Mon Aug 22 01:51:49 2011
#579: https://github.com/saptman/sympy permutation_groups
      Author   : "Saptarshi Mandal" sapta.iitkgp@gmail.com
      Date     : Sat Aug 27 07:19:07 2011
#618: https://github.com/rlamy/sympy Symbol.is_commutative
      Author   : "Ronan Lamy" <>
      Date     : Sat Sep 24 19:13:50 2011
#622: https://github.com/Narsil/sympy master
      Author   : "unknown" <>
      Date     : Tue Sep 27 08:33:55 2011
#673: https://github.com/Krastanov/sympy plotting
      Author   : "Stefan Krastanov" krastanov.stefan@gmail.com
      Date     : Fri Oct 21 16:10:59 2011
#677: https://github.com/mamueller/sympy jordanMatrix
      Author   : "unknown" <>
      Date     : Mon Oct 24 09:54:44 2011
#740: https://github.com/raphaelm/sympy german_tutorial
      Author   : "Raphael Michel" webmaster@raphaelmichel.de
      Date     : Wed Nov 23 07:12:34 2011
#747: https://github.com/hrickards/sympy see_also_integrals
      Author   : "Harry Rickards" hrickards@gmail.com
      Date     : Wed Nov 23 18:27:55 2011
#764: https://github.com/101man/sympy tutorial_french
      Author   : "unknown" <>
      Date     : Sun Nov 27 15:48:17 2011
#785: https://github.com/Slasher9/sympy patch-1
      Author   : "unknown" <>
      Date     : Tue Nov 29 10:52:03 2011
#798: https://github.com/Gentoolx/sympy gci2011-task_7130389
      Author   : "Adrian Borucki" <>
      Date     : Thu Dec 01 21:07:05 2011
#801: https://github.com/benglard/sympy benglard
      Author   : "Benjamin Englard" <>
      Date     : Fri Dec 02 00:12:51 2011
#805: https://github.com/rlamy/sympy pytest
      Author   : "Ronan Lamy" <>
      Date     : Fri Dec 02 05:31:10 2011
#911: https://github.com/toolforger/sympy issue-2867-lambdify-raises
      Author   : "Joachim Durchholz" jo@durchholz.org
      Date     : Fri Dec 30 15:20:00 2011
#914: https://github.com/raoulb/sympy trigonometric
      Author   : "unknown" <>
      Date     : Sat Dec 31 19:01:59 2011
#918: https://github.com/siddhanathan/sympy Sphinx
      Author   : "Siddhanathan Shanmugam" <>
      Date     : Sun Jan 01 17:33:55 2012

---

Patches that can be merged without conflicts:
#525: https://github.com/sherjilozair/sympy linalg3
      Author   : "Sherjil Ozair" sherjilozair@gmail.com
      Date     : Tue Aug 02 19:35:35 2011
#531: https://github.com/saptman/sympy combinatorics
      Author   : "Saptarshi Mandal" sapta.iitkgp@gmail.com
      Date     : Sat Aug 06 15:56:30 2011
#533: https://github.com/saptman/sympy ntheory
      Author   : "Saptarshi Mandal" sapta.iitkgp@gmail.com
      Date     : Sat Aug 06 22:35:08 2011
#609: https://github.com/pernici/sympy lpoly2
      Author   : "unknown" <>
      Date     : Sun Sep 18 16:18:10 2011
#690: https://github.com/smichr/sympy xsub
      Author   : "Christopher Smith" <>
      Date     : Sun Oct 30 10:23:20 2011
#715: https://github.com/asmeurer/sympy 2021
      Author   : "Aaron Meurer" asmeurer@gmail.com
      Date     : Mon Nov 14 16:57:29 2011
#722: https://github.com/goodok/sympy flatten_sort
      Author   : "Alexey U. Gudchenko" <>
      Date     : Sat Nov 19 01:14:25 2011
#726: https://github.com/vperic/sympy use2to3-fixes
      Author   : "Vladimir Peri─ç" <>
      Date     : Mon Nov 21 13:55:21 2011
#786: https://github.com/pk1k/sympy tutorial_pl
      Author   : "Piotr Korgul" <>
      Date     : Tue Nov 29 16:53:00 2011
#793: https://github.com/sognat/sympy issue-2513
      Author   : "Grzegorz ┼Üwirski" sognat@gmail.com
      Date     : Wed Nov 30 21:18:48 2011
#795: https://github.com/dvlahovski/sympy my_trigsimp
      Author   : "unknown" <>
      Date     : Thu Dec 01 09:34:21 2011
#797: https://github.com/goodok/sympy logo
      Author   : "Alexey U. Gudchenko" <>
      Date     : Thu Dec 01 14:51:01 2011
#819: https://github.com/Gentoolx/sympy gci2011-task_7193396
      Author   : "Adrian Borucki" <>
      Date     : Mon Dec 05 17:47:16 2011
#838: https://github.com/smichr/sympy inv-check
      Author   : "Christopher Smith" <>
      Date     : Sat Dec 10 17:37:27 2011
#874: https://github.com/flacjacket/sympy coupling_tests
      Author   : "Sean Vig" <>
      Date     : Wed Dec 21 20:26:45 2011
#875: https://github.com/smichr/sympy cse
      Author   : "Christopher Smith" <>
      Date     : Thu Dec 22 02:16:22 2011
#881: https://github.com/asmeurer/sympy pylint_jcreus
      Author   : "Aaron Meurer" asmeurer@gmail.com
      Date     : Fri Dec 23 17:13:29 2011
#892: https://github.com/goodok/sympy ru_tut
      Author   : "Alexey U. Gudchenko" <>
      Date     : Mon Dec 26 19:31:19 2011
#899: https://github.com/toolforger/sympy issue-1740
      Author   : "Joachim Durchholz" jo@durchholz.org
      Date     : Tue Dec 27 22:49:26 2011
#902: https://github.com/mrocklin/sympy rv2
      Author   : "Matthew Rocklin" <>
      Date     : Wed Dec 28 05:50:38 2011
#903: https://github.com/Krom17/sympy master
      Author   : "Christophe Saint-Jean" <>
      Date     : Wed Dec 28 13:14:50 2011
#904: https://github.com/flacjacket/sympy spin_xfails
      Author   : "Sean Vig" <>
      Date     : Thu Dec 29 03:20:37 2011
#908: https://github.com/flacjacket/sympy quantum_printing_tests
      Author   : "Sean Vig" <>
      Date     : Fri Dec 30 00:38:53 2011
#931: https://github.com/smichr/sympy 2957
      Author   : "Christopher Smith" <>
      Date     : Wed Jan 04 11:13:16 2012
#935: https://github.com/goodok/sympy doctest_unicode
      Author   : "Alexey U. Gudchenko" <>
      Date     : Thu Jan 05 11:53:28 2012
#936: https://github.com/arpit8991/sympy refactorga
      Author   : "unknown" <>
      Date     : Thu Jan 05 21:00:17 2012
#937: https://github.com/Habel/sympy doctests
      Author   : "Matt Habel" habelinc@gmail.com
      Date     : Fri Jan 06 19:49:27 2012
#938: https://github.com/raoulb/sympy fresnel
      Author   : "unknown" <>
      Date     : Fri Jan 06 20:37:15 2012
#939: https://github.com/jerryma1121/sympy workspace
      Author   : "unknown" <>
      Date     : Sat Jan 07 03:38:06 2012

**Comment by goodok** [comment]
Ok. It is work as expected. The output of the 'sympy-bot -n review' is large and will be not necessary. The main thing that the script do not stop with exception.

**Comment by smichr** [comment]
It was just finishing as follows:

$ c:/python27/python.exe sympy-bot review -n 938

> Working directory: c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v
> Reviewing pull request #938
> Pull request info:
>     Author: "unknown" <>
>     Repository: https://github.com/raoulb/sympy
>     Branch: fresnel
> Cloning sympy/sympy master
> Cloning into sympy...
> remote: Counting objects: 60262, done.
> remote: Compressing objects: 100% (14945/14945), done.
> remote: Total 60262 (delta 47871), reused 56777 (delta 44786)
> Receiving objects: 100% (60262/60262), 34.71 MiB | 125 KiB/s, done.
> Resolving deltas: 100% (47871/47871), done.
> Testing...
> Running tests with the following setup:
> master_repo_url = https://github.com/sympy/sympy.git
> pull_request_repo_url = https://github.com/raoulb/sympy
> pull_request_branch = fresnel
> master_repo_path =
> c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/sympy
> test_command = setup.py test
> interpreter = python
> cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/sympy && git
> fetch htt
> ps://github.com/raoulb/sympy fresnel:test
> remote: Counting objects: 49, done.
> remote: Compressing objects: 100% (17/17), done.
> remote: Total 43 (delta 34), reused 35 (delta 26)
> Unpacking objects: 100% (43/43), done.
> From https://github.com/raoulb/sympy
> - [new branch]      fresnel    -> test
>   cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/sympy && git
>   checkout
>   test
>   Switched to branch 'test'
>   Running unit tests.
>   Command: cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/sympy &&
>   git
>   merge origin/master
>   Merge made by recursive.
>   sympy/combinatorics/generators.py   |    8 ++
>   sympy/combinatorics/graycode.py     |   43 ++++++++--
>   sympy/combinatorics/permutations.py |  158
>   ++++++++++++++++++++++++++++++++---
>   sympy/combinatorics/prufer.py       |    8 ++
>   sympy/combinatorics/subsets.py      |   72 ++++++++++++++++
>   sympy/printing/latex.py             |    8 +-
>   sympy/printing/tests/test_latex.py  |    7 ++-
>   7 files changed, 281 insertions(+), 23 deletions(-)
>   Running unit tests.
>   Command: cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/sympy &&
>   pyth
>   on setup.py test
>   'python' is not recognized as an internal or external command,
>   operable program or batch file.
>   Return code: 1
>   Done.

View log at: c:\users\leslie\appdata\local\temp\sympy-bot-tmp_pfo5v/out/log
The syntax of the command is incorrect.
Traceback (most recent call last):
  File "sympy-bot", line 316, in <module>
    main()
  File "sympy-bot", line 126, in main
    review(n, options, username, password)
  File "sympy-bot", line 284, in review
    cmd("mkdir -p %s/out" % tmpdir)
  File "c:\Users\leslie\sympy\sympy-bot\utils.py", line 49, in cmd
    raise CmdException("Command '%s' failed with err=%d." % (s, r))
utils.CmdException: Command 'mkdir -p
c:\users\leslie\appdata\local\temp\sympy-b
ot-tmp_pfo5v/out' failed with err=1.

/c

**Comment by goodok** [comment]
Ok. Thank you.

I see that the 'c:\python27' is not at your PATH environment, so the system don't know about 'python' command.
If you have no reasons against it you can add python directory 'c:\python27' to PATH as described in:
http://www.computerhope.com/issues/ch000549.htm

Then (after shell reloading) you can use just 'python' instead of 'c:\python27' in command shell.
And, I hope, the sympy-bot further will not stopped as it was in your above output.

**Comment by smichr** [comment]
On Sat, Jan 7, 2012 at 10:43 PM, Alexey U. Gudchenko <
reply@reply.github.com

> wrote:
> 
> Ok. Thank you.
> 
> I see that the 'c:\python27' is not at your PATH environment, so the
> system don't know about 'python' command.
> If you have no reasons against it you can add python directory
> 'c:\python27' to PATH as described in:
> http://www.computerhope.com/issues/ch000549.htm
> 
> Then (after shell reloading) you can use just 'python' instead of
> 'c:\python27' in command shell.
> And, I hope, the sympy-bot further will not stopped as it was in your
> above output.
> 
> It appears to be running now:

$ python.exe sympy-bot review -n 938

> Working directory: c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn
> Reviewing pull request #938
> Pull request info:
>     Author: "unknown" <>
>     Repository: https://github.com/raoulb/sympy
>     Branch: fresnel
> Cloning sympy/sympy master
> Cloning into sympy...
> remote: Counting objects: 60262, done.
> remote: Compressing objects: 100% (14946/14946), done.
> remote: Total 60262 (delta 47870), reused 56778 (delta 44785)
> Receiving objects: 100% (60262/60262), 34.71 MiB | 107 KiB/s, done.
> Resolving deltas: 100% (47870/47870), done.
> Testing...
> Running tests with the following setup:
> master_repo_url = https://github.com/sympy/sympy.git
> pull_request_repo_url = https://github.com/raoulb/sympy
> pull_request_branch = fresnel
> master_repo_path =
> c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/sympy
> test_command = setup.py test
> interpreter = python
> cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/sympy && git
> fetch htt
> ps://github.com/raoulb/sympy fresnel:test
> remote: Counting objects: 61, done.
> remote: Compressing objects: 100% (27/27), done.
> remote: Total 54 (delta 41), reused 40 (delta 27)
> Unpacking objects: 100% (54/54), done.
> From https://github.com/raoulb/sympy
> - [new branch]      fresnel    -> test
>   cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/sympy && git
>   checkout
>   test
>   Switched to branch 'test'
>   Running unit tests.
>   Command: cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/sympy &&
>   git
>   merge origin/master
>   Merge made by recursive.
>   sympy/combinatorics/generators.py   |    8 ++
>   sympy/combinatorics/graycode.py     |   43 ++++++++--
>   sympy/combinatorics/permutations.py |  158
>   ++++++++++++++++++++++++++++++++---
>   sympy/combinatorics/prufer.py       |    8 ++
>   sympy/combinatorics/subsets.py      |   72 ++++++++++++++++
>   sympy/printing/latex.py             |    8 +-
>   sympy/printing/tests/test_latex.py  |    7 ++-
>   7 files changed, 281 insertions(+), 23 deletions(-)
>   Running unit tests.
>   Command: cd c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/sympy &&
>   pyth
>   on setup.py test
>   running test
>   ============================= test process starts
>   ==============================

executable:   c:\Python26\python.exe  (2.6.6-final-0)
architecture: 32-bit
cache:        yes
ground types: python
random seed:  69154428

sympy\assumptions\tests\test_assumptions_2.py[5] .....
 [OK]

sympy\assumptions\tests\test_query.py[39]
..ff......w...........................

.
[OK]

sympy\assumptions\tests\test_refine.py[4] ....
 [OK]

sympy\combinatorics\tests\test_generators.py[1] .
[OK]

sympy\combinatorics\tests\test_graycode.py[1] .
[OK]

sympy\combinatorics\tests\test_permutations.py[4] ....
 [OK]

sympy\combinatorics\tests\test_prufer.py[2] ..
 [OK]

sympy\combinatorics\tests\test_subsets.py[1] .
 [OK]

sympy\concrete\tests\test_gosper.py[11] .......f...
[OK]

sympy\concrete\tests\test_products.py[7] .......
 [OK]

sympy\concrete\tests\test_sums_products.py[28] ...fff......................
[OK]

sympy\core\tests\test_args.py[322]
FfffffffXff...............sssssssssssssssssss

**Comment by smichr** [comment]
The test ended like this:

Exception in thread Thread-20:
Traceback (most recent call last):
  File "c:\Python26\lib\threading.py", line 532, in **bootstrap_inner
    self.run()
  File "c:\Python26\lib\threading.py", line 484, in run
    self.__target(_self.__args, *_self.__kwargs)
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\plott
ing\plot_mode_base.py", line 269, in _calculate_all
    self._calculate_cverts()
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\plott
ing\plot_mode_base.py", line 289, in _calculate_cverts
    self._on_calculate_cverts()
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\plott
ing\plot_curve.py", line 53, in _on_calculate_cverts
    inc_pos=inc_work_pos)
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\plott
ing\color_scheme.py", line 261, in apply_to_curve
    cverts[_u][_c])
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\plott
ing\util.py", line 105, in rinterpolate
    return (a_value - a_min) / float(a_range)
  File
"c:\Users\leslie\AppData\Local\Temp\sympy-bot-tmp8wnlwn\sympy\sympy\core\
expr.py", line 161, in __float**
    raise ValueError("Symbolic value, can't compute")
ValueError: Symbolic value, can't compute

Return code: 1
Done.

View log at: c:\users\leslie\appdata\local\temp\sympy-bot-tmp8wnlwn/out/log
The syntax of the command is incorrect.
Traceback (most recent call last):
  File "sympy-bot", line 316, in <module>
    main()
  File "sympy-bot", line 126, in main
    review(n, options, username, password)
  File "sympy-bot", line 284, in review
    cmd("mkdir -p %s/out" % tmpdir)
  File "c:\Users\leslie\sympy\sympy-bot\utils.py", line 49, in cmd
    raise CmdException("Command '%s' failed with err=%d." % (s, r))
utils.CmdException: Command 'mkdir -p
c:\users\leslie\appdata\local\temp\sympy-b
ot-tmp8wnlwn/out' failed with err=1.

File: testrunner.py [hunk 1]
```diff
@@ -43,35 +43,35 @@ def run_tests(master_repo_url, pull_request_repo_url, pull_request_branch,
...
     if r != 0:
-        conflicts = cmd("cd %s; git --no-pager diff" % master_repo_path,
+        conflicts = cmd("cd %s && git --no-pager diff" % master_repo_path,
                 capture=True)
         result["result"] = "conflicts"
         result["log"] = merge_log + "\nLIST OF CONFLICTS\n" + conflicts
         return result
     if python3:
-        cmd("cd %s; bin/use2to3" % master_repo_path)
+        cmd("cd %s && python bin/use2to3" % master_repo_path)
```
Comment (goodok) [inline testrunner.py:72]: I am not sure in this line.
To run 'bin/use2to3' upon windows we have to use some interpreter.
This line assume that 'python' is present in PATH environment variable. But may be we must use python executable explicitly, with which the  sympy-bot is started.

**Comment by goodok** [comment]
I have updated the script and test it now.

Note that for windows it is needed to have 'python' and 'python3' (.bat or .exe) in the PATH variable.
'python' - for running 'bin/use2to3', and the python3 for running python 3.

This situation is not good, but corresponds to the situation upon linux.

**Comment by goodok** [comment]
Have been tested for the Linux, Windows on the example of [sympy/pull/943](https://github.com/sympy/sympy/pull/943).
But not tested for Darwin.

**Comment by asmeurer** [comment]
I'm using this to test sympy/sympy#543.

**Comment by asmeurer** [comment]
I'm going to merge this.  Please open a new issue if there were any issues not addressed, or a new issue created by this.

**Pull request merged**